### ISSUE TITLE
Registration captchas

### ISSUE THREAD

**Issue by mouse-reeve** [bookwyrm-social/bookwyrm#1418]
Let admins enable a captcha to help mitigate spam registrations. The simplest option would be to use google's recaptcha: https://www.google.com/recaptcha/about/, but it would be great to support an independent and/or source-available alternative if that's a viable alternative.

Requirements:
 - [ ] Effective at detecting bot signups
 - [ ] Low-friction for registering users
 - [ ] Stable and actively maintained
 - [ ] Accessible

@oragegu I completely forgot about this as an anti-spam option until just now, and I'd love your thoughts on it

**Comment by mouse-reeve** [comment]
There's a w3c [list of captcha options](https://www.w3.org/WAI/GL/wiki/Captcha_Alternatives_and_thoughts), but Google recaptcha seems like the clear best option from that list.

**Comment by oragegu** [comment]
> There's a w3c [list of captcha options](https://www.w3.org/WAI/GL/wiki/Captcha_Alternatives_and_thoughts), but Google recaptcha seems like the clear best option from that list.

I think custom questions are largely sufficient. Captchas are nice but dispensable.

**Comment by oragegu** [comment]
Besides, captchas are not very friendly towards the visually impaired people.

**Comment by mouse-reeve** [comment]
Yeah, I'm definitely not adding a captcha unless it's seamless for screenreader users, so that will be something to test, and this would be an opt-in feature for admins. Thanks for weighing in -- I'll prioritize adding custom fields for the time being.

**Comment by oragegu** [comment]
> Yeah, I'm definitely not adding a captcha unless it's seamless for screenreader users, so that will be something to test, and this would be an opt-in feature for admins. Thanks for weighing in -- I'll prioritize adding custom fields for the time being.

Besides inclusivity, there is also this question of what's your vision for bookwyrm. Do you picture there be massive instances with loads of users? Then in that case captchas might be useful. 

But if your vision consists of mostly small instances:
-  Captchas might encourage instances to grow bigger and bigger
- Some admins might just get lazy and use captcha to filter people out

**Comment by mouse-reeve** [comment]
I do want to facilitate small instances over big ones, but I think I'm viewing it in kind of the opposite way that you are. My thinking is that the more I work I can take off the hands of admins (in terms of avoiding spam and bots, among other things), the easier it will be for people to take on the task of running an instance (which is hard! it's a lot of work!).

It's definitely true that admins manually approving new users, or reviewing signups, has a lot of benefits for keeping communities small and close-knit, but I think making spam-fighting a more labor intensive process isn't the best way to facilitate that. If its working properly, a captcha wouldn't be filtering people out so much as filtering out scripted bot registration, so I think of it as solving a different problem than managing the community of people.

That said, I'm glad your thinking about this from the perspective of running a small instance and what works best for that! Since I'm managing a large instance, it's easy for me to think in a mindset of what works at that scale, rather than at a more comfortable size. I'll keep thinking about this, and other ways to support and encourage small instances.
